rust/lance-index/src/vector/flat/index.rs

```diff
@@ -134,6 +135,10 @@ impl IvfSubIndex for FlatIndex {
         Ok(Self {})
     }
 
+    fn remap(&self, _: &HashMap<u64, Option<u64>>) -> Result<Self> {
+        Ok(self.clone())
```

Comment:
nit: let's add a warning log here?

Reply:
oh w8, we should remap sub index here no?

Reply:
for v3 we need to remap the subindex & vector storage. flat index doesn't contain anything so it simply returns itself here

Reply:
flat map is still `{ origin_vector: row_id }`? if `row id` changes during compaction, we need to remap them ?

Reply:
remap on an vector index (v3) is:
- remap the sub index
- remap the storage
for IVF_FLAT, the sub index is `FLAT` and storage is `FlatStorage`. FLAT sub index doesn't contain any data so no need to do anything here. the remapping happens on `FlatStorage`